### PR TITLE
[v12] ec2discovery: Use a random name for the install script

### PR DESCRIPTION
**Pull request by jentfoo**
Backport #35152 to branch/v12

### CODE REVIEW

**Review by codingllama — APPROVED** [review]

File: lib/configurators/aws/aws_test.go [hunk 1]
```diff
@@ -1317,6 +1318,13 @@ type SSMMock struct {
 func (m *SSMMock) CreateDocumentWithContext(ctx aws.Context, input *ssm.CreateDocumentInput, opts ...request.Option) (*ssm.CreateDocumentOutput, error) {
 
 	m.t.Helper()
+
+	// UUID's are unpredictable, so we remove them from the content
```
[the remark this answers is not in the document]
Comment (jentfoo) [reply lib/configurators/aws/aws_test.go:1322]: The comparison of the document would never be possible without this replacement.  So I believe this logic should be colocated wherever we are doing the `require.Equals` check.

[the remark this answers is not in the document]
Comment (codingllama) [reply lib/configurators/aws/aws_test.go:1322]: Nevermind, I thought the require.Equal was somewhere else, as I was not expecting the mock to hold a `testing.T`.

File: lib/configurators/aws/aws_test.go [hunk 2]
```diff
@@ -1317,6 +1318,13 @@ type SSMMock struct {
...
+	uuidRegex := regexp.MustCompile(`[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}`)
+	replacedExpected := uuidRegex.ReplaceAllString(*m.expectedInput.Content, "")
+	m.expectedInput.Content = &replacedExpected
+	replacedInput := uuidRegex.ReplaceAllString(*input.Content, "")
+	input.Content = &replacedInput
 	require.Equal(m.t, m.expectedInput, input)
```
Comment (codingllama) [inline lib/configurators/aws/aws_test.go:1328]: Maybe?

```suggestion
	// Diff content first for a nicer error message.
	require.Empty(m.t,
		cmp.Diff(m.expectedInput.Content, input.Content),
		"Document content diff (-want +got)")
	require.Empty(m.t,
		cmp.Diff(m.expectedInput, input, cmpopts.IgnoreFields(ssm.CreateDocumentInput{}, "Content")),
		"Document diff (-want +got)")
```

cmpopts is github.com/google/go-cmp/cmp/cmpopts

File: lib/configurators/aws/aws_test.go [hunk 3]
```diff
@@ -1317,6 +1318,13 @@ type SSMMock struct {
 func (m *SSMMock) CreateDocumentWithContext(ctx aws.Context, input *ssm.CreateDocumentInput, opts ...request.Option) (*ssm.CreateDocumentOutput, error) {
 
 	m.t.Helper()
+
+	// UUID's are unpredictable, so we remove them from the content
+	uuidRegex := regexp.MustCompile(`[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}`)
```
Comment (codingllama) [inline lib/configurators/aws/aws_test.go:1323]: Cache the regex? Go 1.21 can use [sync.OnceValue](https://pkg.go.dev/sync#OnceValue), older versions can do something like this:

(top-level declaration)

```go
var uuidRegex = struct {
	value *regexp.Regexp
	once  sync.Once
}{}
```

and then:

```go
	uuidRegex.once.Do(func() {
		uuidRegex.value = regexp.MustCompile(`[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}`)
	})
```

**Review by codingllama — APPROVED** [review]

File: lib/configurators/aws/aws_test.go (hunk 3 above)
Comment (jentfoo) [reply lib/configurators/aws/aws_test.go:1323]: Is it worth the (admittedly minor) complexity for testing?

Comment (codingllama) [reply lib/configurators/aws/aws_test.go:1323]: sync.OnceValue is pretty cheap, which is how I started. Your call.

File: lib/configurators/aws/aws_test.go (hunk 2 above)
Comment (jentfoo) [reply lib/configurators/aws/aws_test.go]: Updated, thank you

**Review by greedy52 — APPROVED** [review]

**Pull request merged**